app/pages/admin/project-status.jsx

```diff
@@ -62,6 +70,40 @@ class ProjectStatus extends Component {
       .catch(error => this.setState({ error }));
   }
 
+  onBlurTrainingSetIds(workflow, event) {
+    this.setState({ error: null });
+    let val = this.state.valTrainingSetIds[workflow.id] || '';
+    val = val.split(',').map((str) => { return parseInt(str) })
+      .filter((num) => { return !isNaN(num) });
+    // TODO: if the value is an empty string, the field should be deleted altogether.
+    return workflow.update({ 'configuration.training_set_ids': val }).save()
+      .then(() => this.getWorkflows())
+      .catch(error => this.setState({ error }));
+  }
+
+  onBlurTrainingChances(workflow, event) {
+    this.setState({ error: null });
+    let val = this.state.valTrainingChances[workflow.id] || '';
+    val = val.split(',').map((str) => { return parseFloat(str) })
+      .filter((num) => { return !isNaN(num) });
+    // TODO: if the value is an empty string, the field should be deleted altogether.
+    return workflow.update({ 'configuration.training_chances': val }).save()
+      .then(() => this.getWorkflows())
+      .catch(error => this.setState({ error }));
+  }
+
+  onBlurTrainingDefaultChance (workflow, event) {
+    this.setState({ error: null });
+    const val =
+      this.state.valTrainingDefaultChance[workflow.id]
+      ? parseFloat(this.state.valTrainingDefaultChance[workflow.id]) : '';
```

Comment:
This sets the value to either a number or an empty string. Just checking that the mixed types are intended? I don't see it causing any problems.

Reply:
Yup, that's correct! Accepted values are either numbers, or an empty string.